extensions/wikidata/src/org/openrefine/wikidata/qa/scrutinizers/ItemRequiresScrutinizer.java

```diff
@@ -0,0 +1,102 @@
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ItemRequiresScrutinizer extends EditScrutinizer {
+
+    public static final String newItemType = "item-requires-certain-other-statement";
+    public static final String existingItemType = "item-should-have-certain-other-statement";
+    public static String ITEM_REQUIRES_CONSTRAINT_QID = "Q21503247";
+    public static String ITEM_REQUIRES_PROPERTY_PID = "P2306";
+    public static String ITEM_OF_PROPERTY_CONSTRAINT_PID = "P2305";
+
+    class ItemRequiresConstraint {
+        final PropertyIdValue itemRequiresPid;
+        final List<Value> itemList;
+
+        ItemRequiresConstraint(Statement statement) {
+            List<SnakGroup> specs = statement.getClaim().getQualifiers();
+            PropertyIdValue pid = null;
+            this.itemList = new ArrayList<>();
+            for(SnakGroup group : specs) {
+                for (Snak snak : group.getSnaks()) {
+                    if (group.getProperty().getId().equals(ITEM_REQUIRES_PROPERTY_PID)){
+                        pid = (PropertyIdValue) snak.getValue();
+                    }
+                    if (group.getProperty().getId().equals(ITEM_OF_PROPERTY_CONSTRAINT_PID)){
+                        this.itemList.add(snak.getValue());
+                    }
+                }
+            }
+            this.itemRequiresPid = pid;
+        }
+    }
+
+    @Override
+    public void scrutinize(ItemUpdate update) {
+        Map<PropertyIdValue, Set<Value>> propertyIdValueValueMap = new HashMap<>();
+        for (Statement statement : update.getAddedStatements()) {
+            PropertyIdValue pid = statement.getClaim().getMainSnak().getPropertyId();
+            Value value = statement.getClaim().getMainSnak().getValue();
+            Set<Value> values;
+            if (value != null) {
+                if (propertyIdValueValueMap.containsKey(pid)) {
+                    values = propertyIdValueValueMap.get(pid);
+                } else {
+                    values = new HashSet<>();
+                }
+                values.add(value);
+                propertyIdValueValueMap.put(pid, values);
+            }
+        }
+
+        for (PropertyIdValue propertyId : propertyIdValueValueMap.keySet()) {
+            List<Statement> constraintDefinitions = _fetcher.getConstraintsByType(propertyId, ITEM_REQUIRES_CONSTRAINT_QID);
+            for (Statement statement : constraintDefinitions) {
+                ItemRequiresConstraint constraint = new ItemRequiresConstraint(statement);
+                PropertyIdValue itemRequiresPid = constraint.itemRequiresPid;
+                List<Value> itemList = constraint.itemList;
+                if (!propertyIdValueValueMap.containsKey(itemRequiresPid)) {
+                    QAWarning issue = new QAWarning(update.isNew() ? newItemType : existingItemType, propertyId.getId() + itemRequiresPid.getId(), QAWarning.Severity.WARNING, 1);
```

Comment:
The severity should also be adapted depending on whether the the `update.isNew()` or not

Reply:
Sorry, missed by me.